lib/fastlane/fast_file.rb

```diff
@@ -24,19 +24,38 @@ def initialize(path = nil)
                     'you should turn off smart quotes in your editor of choice.'.red
       end
 
-      parse(content)
+      parse(content, @path)
     end
 
-    def parse(data)
+    def parsing_binding
+      binding
+    end
+
+    def parse(data, path = nil)
       @runner ||= Runner.new
 
       Dir.chdir(Fastlane::FastlaneFolder.path || Dir.pwd) do # context: fastlane subfolder
+        # create nice path that we want to print in case of some problem
+        relative_path = path.nil? ? nil : Pathname.new(path).relative_path_from(Pathname.new(Dir.pwd)).to_s
+
         begin
+          # We have to use #get_binding method, because some test files defines method called `path` (for example SwitcherFastfile)
+          # and local variable has higher priority, so it causes to remove content of original Fastfile for example. With #get_binding
+          # is this always clear and safe to declare any local variables we want, because the eval function uses the instance scope
+          # instead of local.
+
           # rubocop:disable Lint/Eval
-          eval(data) # this is okay in this case
+          if path.nil?
```

Comment:
We have to, because it is trying to convert nil to String otherwise. I guess it detects number of arguments passing to `eval` function. I can change this to something like this:

```ruby
params = [data, parsing_binding, relative_path].compact
eval(*params)
```

Is this better?